docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,669 @@
+
+1. Example payload
+    ```json
+   "receiver": "webhook_receiver",
+   "status": "firing"
+   "alerts": [
+        {
+            "status": "firing",
+            "labels": {
+                "alertname": "UpdateAvailable",
+                "channel": "stable-4.16",
+                "managed_cluster": "89070983-a62f-4dbe-9457-7e0c27832c63",
+                "namespace": "openshift-cluster-version",
+                "openshift_io_alert_source": "platform",
+                "prometheus": "openshift-monitoring/k8s",
+                "severity": "info",
+                "upstream": "<default>"
+            },
+            "annotations": {
+                "description": "For more information refer to 'oc adm upgrade' or https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/settings/cluster/.",
+                "summary": "Your upstream update recommendation service recommends you update your cluster."
+            },
+            "startsAt": "2024-08-28T11:39:17.958Z",
+            "endsAt": "0001-01-01T00:00:00Z",
+            "generatorURL": "https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/monitoring/graph?g0.expr=sum+by+%28channel%2C+namespace%2C+upstream%29+%28cluster_version_available_updates%29+%3E+0&g0.tab=1",
+            "fingerprint": "91406cd113ad87e5"
+        },
+   ]
+    ```
+2. Sync `alarm_event_record` Table
+   - Update rows to "resolved" that are missing in the current payload (i.e previously seen but somehow missed the "resolved" notification)
+     ```sql
+      UPDATE alarm_event_record
+      SET status = 'resolved', alarm_cleared_time = CURRENT_TIMESTAMP
+      WHERE (finger_print, alarm_raised_time) NOT IN (
+             ('Something-Thats-Only-Now-Available', '2023-09-01 10:02:00+00')
+      );
+     ```
+   - Create new AlarmEventRecord. Alert entry payload will require Alarm Definition ID and Probable Cause ID which can retrieved as shown [here](#for-a-given-resourcetypeid-and-alarmname-coming-from-am-alert-find-the-alarmdefinitionid-and-probablecauseid)
+   - Upsert all AlarmEventRecord all the "firing" and "resolved" alerts as indicated by `alerts[].status` i.e a bulk insert + update operation 
+     ```sql
+      -- ...INSERT, follwed by...
+      ON CONFLICT ON CONSTRAINT unique_finger_print_alarm_raised_time DO UPDATE -- defined in schema 
+      SET status = EXCLUDED.status,
+      alarm_cleared_time = EXCLUDED.alarm_cleared_time,
+      alarm_changed_time = EXCLUDED.alarm_changed_time;
+     ```
+3. Grab the Subscriptions and send notification. 
+   - Database interaction is further explained [here](#notification-tracking)
+4. Move all the `status: resolved` rows from `alarm_event_record` to `alarm_event_record_archive`
+
+Eventually data in `alarm_event_record_archive` will be cleared (hardcoded to 24hr) as seen [here](#daily-archive-cleanup-)
+
+## Schema
+We only take care of Alarms* data contained within a specific DB, this approach allows for
```

Comment:
I don't understand what that means - can you elaborate or provide an example?

Reply:
Yeah it's a bit confusing. Basically it's just to each service that needs a DB will create their own. 

e.g alarm server will run something 
```
CREATE DATABASE o-ran-infrastructure-monitoring-alarms;
``` 
a "myservice" will run 
```
CREATE DATABASE myservice;
``` 

From their own migration/init scripts

Reply:
Please clarify in the doc that you are creating the alarm service DB in the O-Cloud DB service

Reply:
updated to clarify